lib/tarpon/request/base.rb

```diff
@@ -10,14 +11,19 @@ class Base
         content_type: 'application/json'
       }.freeze
 
+      def initialize(client: Client.default)
```

Comment:
We can explicitly require a `client` to be passed and stop relying on the `Client` class for that.
This prevents weird behavior with eventual overwrites.
```suggestion
      def initialize(client:)
```